src/main/scala/ai/privado/languageEngine/java/tagger/sink/JavaAPITagger.scala

```diff
@@ -0,0 +1,129 @@
+
+import ai.privado.cache.AppCache
+import ai.privado.languageEngine.java.language.{NodeStarters, NodeToProperty, StepsForProperty}
+import ai.privado.metric.MetricHandler
+import ai.privado.model.{Constants, Language, RuleInfo}
+import ai.privado.tagger.PrivadoSimplePass
+import ai.privado.utility.ImportUtility
+import ai.privado.utility.Utilities.{
+  addRuleTags,
+  getDefaultSemantics,
+  getFileNameForNode,
+  isFileProcessable,
+  storeForTag
+}
+import io.circe.Json
+import io.joern.dataflowengineoss.language._
+import io.joern.dataflowengineoss.queryengine.{EngineConfig, EngineContext}
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.CfgNode
+import io.shiftleft.semanticcpg.language._
+import overflowdb.BatchedUpdate
+
+import scala.collection.mutable
+
+class JavaAPITagger(cpg: Cpg) extends PrivadoSimplePass(cpg) {
+
+  val cacheCall                  = cpg.call.where(_.nameNot("(<operator|<init).*")).l
+  val internalMethodCall         = cpg.method.dedup.isExternal(false).fullName.take(30).l
+  val topMatch                   = mutable.HashMap[String, Integer]()
+  val COMMON_IGNORED_SINKS_REGEX = "(?i).*(?<=map|list|jsonobject|json|array|arrays).*(put:|get:).*"
+  lazy val APISINKS_REGEX =
+    "(?i)(?:url|client|openConnection|request|execute|newCall|load|host|access|fetch|get|getInputStream|getApod|getForObject|getForEntity|list|set|put|post|proceed|trace|patch|Path|send|" +
+      "sendAsync|remove|delete|write|read|assignment|provider|exchange|postForEntity)"
+
+  internalMethodCall.foreach((method) => {
+    val key     = method.split("[.:]").take(2).mkString(".")
+    val currVal = topMatch.getOrElse(key, 0).asInstanceOf[Int]
+    topMatch(key) = (currVal + 1)
+  })
+  var APISINKS_IGNORE_REGEX = "^("
+  topMatch.foreach((mapEntry) => {
+    if (mapEntry == topMatch.last) {
+      APISINKS_IGNORE_REGEX += mapEntry._1 + ").*"
+    } else {
+      APISINKS_IGNORE_REGEX += mapEntry._1 + "|"
+    }
+  })
+  val apis = cacheCall
+    .name(APISINKS_REGEX)
+    .methodFullNameNot(APISINKS_IGNORE_REGEX)
+    .methodFullNameNot(COMMON_IGNORED_SINKS_REGEX)
+    .l
+
+  implicit val engineContext: EngineContext = EngineContext(semantics = getDefaultSemantics, config = EngineConfig(4))
+  val commonHttpPackages: String =
+    "^(?i)(org.apache.http|okhttp|org.glassfish.jersey|com.mashape.unirest|java.net.http|java.net.URL|org.springframework.(web|core.io)|groovyx.net.http|org.asynchttpclient|kong.unirest.java|org.concordion.cubano.driver.http|javax.net.ssl).*"
+  override def run(builder: BatchedUpdate.DiffGraphBuilder): Unit = {
+    val apiInternalSinkPattern = cpg.literal.code("(?:\"|')(" + ruleInfo.combinedRulePattern + ")(?:\"|')").l
+    val propertySinks          = cpg.property.filter(p => p.value matches (ruleInfo.combinedRulePattern)).usedAt.l
+    apis.methodFullName(s"${commonHttpPackages}${APISINKS_REGEX}.*").l.size match {
```

Comment:
@abhstabs This decision of using which approach, has to be outside of `run`, because `run` is executed the based on the number of rules, in our case 2 times. 

A `lazy val approachToUse` can be defined a member of `APITagger` which will hold the approach to be used, and in `run` `approachToUse` can be checked